plugin/serve.go

```diff
@@ -7,11 +7,15 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
-// The constants below are the names of the plugins that can be dispensed
-// from the plugin server.
 const (
+	// The constants below are the names of the plugins that can be dispensed
+	// from the plugin server.
 	ProviderPluginName    = "provider"
 	ProvisionerPluginName = "provisioner"
+
+	// DefaultProtocolVersion is the default protocol version to configure
+	// clients and server
+	DefaultProtocolVersion = 5
```

Comment:
This one needs to stay as `4` so that the providers we build are compatible with Terraform 0.10 and 0.11. Terraform 0.12 overrides this during its plugin handshake, so this value won't be used when the client is 0.12.

Reply:
ahh i misunderstood the discussion slightly. thank you for catching this :)